### PR TITLE
 Changed DocumentRoot for Mission Portal

### PR DESCRIPTION
**Pull request by aleksandrychev**
ChangeLog: Changed DocumentRoot for Mission Portal in httpd.conf. New path is /path/to/cfengine/httpd/htdocs/public.

Ticket: ENT-5372

Merge together: 
https://github.com/cfengine/masterfiles/pull/1698
https://github.com/cfengine/buildscripts/pull/694
https://github.com/cfengine/nova/pull/1633
https://github.com/cfengine/mission-portal/pull/1056

### CODE REVIEW

**Comment by aleksandrychev** [comment]
@nickanderson please review

File: cfe_internal/enterprise/CFE_hub_specific.cf [hunk 1]
```diff
@@ -342,7 +342,7 @@ bundle agent cfe_internal_truncate_events
 {
   commands:
     Hr05.Min00_05::
-      "$(sys.workdir)/httpd/php/bin/php $(cfe_internal_hub_vars.docroot)/index.php cli_tasks truncate_events"
+      "$(sys.workdir)/httpd/php/bin/php $(cfe_internal_hub_vars.docroot)/public/index.php cli_tasks truncate_events"
```
Comment (nickanderson) [inline cfe_internal/enterprise/CFE_hub_specific.cf]: Won't this break bookmarks to links inside Mission Portal?

I am thinking things like URLs with base64 encoded sql queries, saved reports, links to individual host pages.

Comment (aleksandrychev) [reply cfe_internal/enterprise/CFE_hub_specific.cf]: URLs won't be changed, this is CLI command that requires path to index.php.

Comment (nickanderson) [reply cfe_internal/enterprise/CFE_hub_specific.cf]: Ok, thanks.

File: cfe_internal/enterprise/CFE_hub_specific.cf [hunk 2]
```diff
@@ -38,6 +38,10 @@ bundle common cfe_internal_hub_vars
         comment => "Root directory of Enterprise Web interface",
         handle => "cfe_internal_hub_vars_docroot";
 
+      "public_docroot" string => ifelse( isdir("$(sys.workdir)/httpd/htdocs/public"), "$(sys.workdir)/httpd/htdocs/public",
+                                               "$(sys.workdir)/httpd/htdocs"),
```
Comment (nickanderson) [inline cfe_internal/enterprise/CFE_hub_specific.cf]: Instead of `"$(sys.workdir)/httpd/htdocs"),` how about using the `docroot` variable, `"$(docroot)"),`

Comment (aleksandrychev) [reply cfe_internal/enterprise/CFE_hub_specific.cf]: @nickanderson  done

**Comment by nickanderson** [comment]
Looks ok to me now (just needs squashed before merging), thanks for fixing the variable and making it backwards compatible.

**Comment by aleksandrychev** [comment]
@nickanderson please merge

**Pull request merged**